src/Controller/ComponentRegistry.php

```diff
@@ -130,6 +130,7 @@ protected function _throwMissingClassError(string $class, ?string $plugin): void
     protected function _create(object|string $class, string $alias, array $config): Component
     {
         if (is_object($class)) {
+            /** @var \Cake\Controller\Component */
```

Comment:
I am surprised this throws an error since the param tag has the same info.

Reply:
The param tag should probably be `\Cake\Controller\Component|class-string<\Cake\Controller\Component>` right?

Reply:
Can we use class-string already in the default tags?
Otherwise we might have to use specific tags on top.

Reply:
We are using it in Debugger in 4.next and a couple other places in 5.x. Not sure if we made the decision for a specific version.

@ADmad Did we decide it's ok?

Reply:
Phpstorm supports `class-string` now, so I don't see any problem using it.

Reply:
Sounds like cs sniffer also is fine then with it